lib/spack/spack/test/cmd/flake8.py

```diff
@@ -86,10 +86,11 @@ def test_flake8(parser, flake8_package):
     # Otherwise, the unit tests would fail every time
     # the flake8 tests fail
     args = parser.parse_args([flake8_package])
-
-    flake8(parser, args)
-
+    try:
+        flake8(parser, args)
     # Get even more coverage
-    args = parser.parse_args(['--output', '--root-relative', flake8_package])
-
-    flake8(parser, args)
+        args = parser.parse_args(['--output', '--root-relative',
+                                 flake8_package])
+        flake8(parser, args)
+    except SystemExit:
+        pytest.skip("no flake8 found in path")
```

Comment:
Yeah, I'm still getting familiar with pytest but that looks much better that try-except.